app-text/wordgrinder/wordgrinder-0.7.1.recipe

```diff
@@ -11,35 +11,38 @@ REVISION="1"
 SOURCE_URI="https://github.com/davidgiven/wordgrinder/archive/$portVersion.tar.gz"
 CHECKSUM_SHA256="54085af7963e1f67342bc0b1b20d1ccc75494f2e23d401b601cf3089acea747c"
 
-ARCHITECTURES="!x86_gcc2 !x86_64"
+ARCHITECTURES="!x86_gcc2 x86 ?x86_64"
+SECONDARY_ARCHITECTURES="x86"
 
 PROVIDES="
-	wordgrinder = $portVersion
-	cmd:WordGrinder = $portVersion
+	wordgrinder$secondaryArchSuffix = $portVersion
+	cmd:wordgrinder$secondaryArchSuffix = $portVersion
 	"
 REQUIRES="
-	haiku
-	lib:libncurses
-	lib:liblua
-	lib:libz
+	haiku$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:liblua$secondaryArchSuffix >= 5.3
+	lib:libz$secondaryArchSuffix
 	"
 
 BUILD_REQUIRES="
-	haiku_devel
-	devel:libncurses
-	devel:liblua
-	devel:libz
+	haiku${secondaryArchSuffix}_devel
+	devel:libncurses$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:liblua$secondaryArchSuffix >= 5.3
+	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
-	cmd:g++
+	cmd:g++$secondaryArchSuffix
 	cmd:make
 	cmd:ninja
-	cmd:pkg_config
+	cmd:pkg_config$secondaryArchSuffix
 	"
 
 BUILD()
 {
-	make OBJDIR=$sourceDir/build PREFIX=$prefix BINDIR=$binDir \
+	make all OBJDIR=$sourceDir/build PREFIX=$prefix BINDIR=$binDir \
```

Comment:
The trick is to define HOME!
BTW, I thought that OBJDIR was needed, but it turns out that we don't need to bother with it.
```diff
 BUILD()
 {
-	make all OBJDIR=$sourceDir/build PREFIX=$prefix BINDIR=$binDir \
+	make PREFIX=$prefix BINDIR=$commandBinDir DOCDIR=$docDir \
-		DOCDIR=$docDir MANDIR=$manDir
+		MANDIR=$manDir HOME=$prefix LUA_PACKAGE=lua5.3
 }

 INSTALL()
 {
-	make all OBJDIR=$sourceDir/build PREFIX=$prefix BINDIR=$binDir \
+	make install PREFIX=$prefix BINDIR=$commandBinDir DOCDIR=$docDir \
-		DOCDIR=$docDir MANDIR=$manDir
+		MANDIR=$manDir HOME=$prefix LUA_PACKAGE=lua5.3
+	mv $docDir/wordgrinder/README.wg $docDir
+	rmdir $docDir/wordgrinder
 }
```
Remark: moving README.wg might have a downside if the program expects it where it thinks it is installed. So maybe we should drop the mv. In that case the README.wg would be in:
* /system/documentation/packages/wordgrinder_x86/wordgrinder/ (for x86_gcc2)
* /system/documentation/packages/wordgrinder/wordgrinder/ (for x86_64)

With the mv it it will  be in:
* /system/documentation/packages/wordgrinder_x86/ (for x86_gcc2)
* /system/documentation/packages/wordgrinder/ (for x86_64)

Reply:
@fbrosson Thanks! That did the trick, only thing I can't seem to set is LUA_PACKAGE=lua5.3
if I just use lua it works :+1:

Reply:
That's strange. It builds for me with both values, lua and lua5.3.
```
Building ...
Bootstrapping build
Windows toolchain: not found
Detecting Curses in package 'ncursesw': found
Detecting FreeType2 in package 'freetype2': not found
Detecting Minizip in package 'builtin': found
Detecting LuaFileSystem in package 'builtin': found
Detecting uthash in package 'builtin': found
Detecting LuaBitOp in package 'builtin': found
Detecting Lua in package 'lua5.3': found
Detecting Lua in package 'builtin': found
Detecting Lua in package 'lua-5.1': not found
Detecting Lua in package 'lua-5.2': not found
Detecting Lua in package 'lua-5.3': not found
Detecting Lua in package 'luajit': not found
The preferred Lua package is: 'lua5.3'
```
Well, let's stick to lua, since it builds for you.

Hmm, I just noticed the `Detecting FreeType2 in package 'freetype2': not found`, so maybe you were right about adding freetype2. I'll try to see if I can have it detected.

I though I had suggested the inclusion of a debuginfo package, but it looks like I forgot about it. If you want to add it, that shoud go afet BUILD_PREREQUIRES and before BUILD:
```
defineDebugInfoPackage wordgrinder$secondaryArchSuffix \
	$commandBinDir/wordgrinder
```

Hmm, could you remove the blank line at the end of the recipe?

**EDIT:** I've just added `lib:libfreetype$secondaryArchSuffix` to REQUIRES, `devel:libfreetype$secondaryArchSuffix` to BUILD_REQUIRES and `cmd:freetype_config$secondaryArchSuffix` to BUILD_PREREQUIRES, and it now finds freetype2:
```
Detecting Curses in package 'ncursesw': found
Detecting FreeType2 in package 'freetype2': found
Detecting Xft in package 'none': not found
Detecting Minizip in package 'builtin': found
Detecting LuaFileSystem in package 'builtin': found
Detecting uthash in package 'builtin': found
Detecting LuaBitOp in package 'builtin': found
Detecting Lua in package 'lua5.3': found
Detecting Lua in package 'builtin': found
```
But then it looks for Xft and does not find it. Shell we dig in that direction?

Reply:
Strange .... I don't see it looking for Xft here (on x86_gcc2)?

Bootstrapping build
  Windows toolchain: not found
  Detecting Curses in package 'ncursesw': found
  Detecting FreeType2 in package 'freetype2': not found
  Detecting Minizip in package 'builtin': found
  Detecting LuaFileSystem in package 'builtin': found
  Detecting uthash in package 'builtin': found
  Detecting LuaBitOp in package 'builtin': found
  Detecting Lua in package 'lua': found
  Detecting Lua in package 'builtin': found
  Detecting Lua in package 'lua-5.1': not found
  Detecting Lua in package 'lua-5.2': not found
  Detecting Lua in package 'lua-5.3': not found
  Detecting Lua in package 'luajit': not found
  The preferred Lua package is: 'lua'